storage/src/mmr/mem.rs

```diff
@@ -35,6 +35,15 @@ pub struct Mmr<H: CHasher> {
 
     // The auxiliary map from node position to the digest of any pinned node.
     pub(super) pinned_nodes: HashMap<u64, H::Digest>,
+
+    // Non-leaf nodes that need to have their digests recomputed due to a batched update operation.
+    //
+    // This is a set of tuples of the form (node_pos, height).
+    dirty_nodes: HashSet<(u64, u32)>,
+
+    // Dummy digest used as a placeholder for nodes whose digests will be updated with the next
+    // `sync`.
+    dirty_digest: H::Digest,
```

Comment:
Still getting comfortable with the approach of putting these fake digests in the vec (although I think it is probably more efficient than using `Option` for each entry or an enum.

Reply:
It always comes down to appeasing the linter (who is smart enough to not let us skip over entires 😓 ).

Reply:
I have come to the conclusion this is the least worst option for this pattern.